packages/clerk-js/src/ui/components/OrganizationSwitcher/UserMembershipList.tsx

```diff
@@ -72,6 +101,7 @@ export const UserMembershipList = (props: UserMembershipListProps) => {
           />
         </PreviewButton>
       ))}
+      {(hasNextPage || isLoading) && <InfiniteListSpinner ref={ref} />}
```

Comment:
❓ are we sure about showing the `InfiniteListSpinner` when `hasNextPage` is true?

Reply:
So we've been following this pattern for a while, i believe this just improves the UI in general by minimizing layout shifts when the loading get triggered.

F.e.: You scroll, u are about to reach the bottom and u see a spinner, the spinner triggers the request (to fetch the next page) and u continue to see the spinner until loading is completed, now the spinner is pushed further now and no requests will fire until it is in the users viewport again.